localstack/services/cloudformation/models/ec2.py

```diff
@@ -339,6 +339,9 @@ def _pre_delete(resource_id, resources, resource_type, func, stack_name):
                 )
                 for rt in resp["RouteTables"]:
                     for assoc in rt.get("Associations", []):
+                        # skipping Main association (accommodating recent upstream change)
```

Comment:
Just a side-node, no need to change it here: The term "recent" might be confusing at a later point in time.

Reply:
Yep, great point - thanks for mentioning that @alexrashed (will include in next iteration..)